### PR TITLE
Refactor Discord class to be reloadable

### PR DESCRIPTION
**Pull request by Salanto**
The Discord Webhook, while convenient, requires a reboot to update some settings of its operation, making it less than ideal for continuous operating in its current state.

This PR resolves these issues by redesigning some aspects of the Webhook to be easier to manage and less reliant on object persistence to keep track of aspects, like the uptime.

TODO :

Configuration file
- [x] Split configuration files for higher visibility
- [x] Update configuration descriptions for increased clarity
- [x] Load webhook related configurations into its own QSetting object

Server
- [x] Move construction/destruction of discord object into its own method
- [x] Cleanup function naming, enforce checks before connecting slots

ConfigManager
- [x] Use QElapsedTimer to keep track of uptime and make it available to other classes

Discord
- [x] Update uptime webhook text and implement the usage of QElapsedTimer

### CODE REVIEW

File: core/src/server.cpp [hunk 1]
```diff
@@ -289,6 +283,27 @@ void Server::allowMessage()
     can_send_ic_messages = true;
 }
 
+void Server::handleDiscordIntegration()
+{
+    if (discord != nullptr) {
```
Comment (MangosArentLiterature) [inline core/src/server.cpp:288]: Why not just add a reload method within discord instead of deleting the entire thing just to reconstruct it?

File: core/src/server.cpp [hunk 2]
```diff
@@ -289,6 +283,27 @@ void Server::allowMessage()
...
+        discord->deleteLater();
```
Comment (MangosArentLiterature) [inline core/src/server.cpp:289]: If you're not going to wait for the deletion, just use delete.

File: core/src/server.cpp (hunk 1 above)
Comment (Salanto) [reply core/src/server.cpp:288]: That would require to include server to disconnect and connect the signals, if it happens in Discord.

My version ensures all ties are servered and I simply do not have to add the logic that is required to recheck all connections and disconnect them if necessary.

Additionally nothing inside of Discord has any  reason to be preserved as it's loaded from ConfigManager on demand.

TLDR : Work in my solution < Work required to handle signals and slots in server.

File: core/src/server.cpp (hunk 2 above)
Comment (Salanto) [reply core/src/server.cpp:289]: ???. I mark it for deletion and invalidate it's pointer in case someone runs /reload twice in a row. Nothing wrong with that. I could listen for the destroyed signal, but that's not really useful.

File: core/src/server.cpp (hunk 1 above)
Comment (MangosArentLiterature) [reply core/src/server.cpp:288]: Nothing in Discord requires it to be destroyed to work properly. As far as I can see, the only thing that's really accomplishing is letting it create that uptime timer. This can literally be done by a function within Discord. handleDiscordIntegration() can call this function, because server has a pointer to discord. There is no point deleting discord just to remake it for this one thing. You don't need to destroy all the signals and slots just to remake them all. If we're doing ban webhooks now when we weren't before, you connect it. If we're no longer doing them, you disconnect it. That's it. You don't have to delete it.

You have a pointer to discord with Server, there is no point to not using it.

File: core/src/server.cpp (hunk 2 above)
Comment (MangosArentLiterature) [reply core/src/server.cpp:289]: Who said anything about listening for a destroyed signal? deleteLater() just posts a deletion event that will be handled later. That basically just calls delete. You're not really waiting for discord to be deleted before you go ahead and remake it, and you're not doing this from a slot Discord can call. It's fine to just delete the object outright here. It's probably trivial, but there's no real benefit to using deleteLater() here as far as I can tell. If that is incorrect in this circumstance, then feel free to ignore me. Though in fairness, I don't think the deletion is necessary at all.

**Review by in1tiate — CHANGES_REQUESTED** [review]
Please do this properly. There's no need to destroy the `discord` object and reconstruct it, all you need to do is start the uptime timer and handle which signal/slot pairs need to be connected or disconnected. Shortcuts like this, while *functional,* decrease the code's maintainability and are a big part of how tsuserver3 got so bad it needed a replacement in the first place.

**Review by in1tiate — APPROVED** [review]
Looks good, thank you.
